addons/survey/static/src/js/survey_form.js

```diff
@@ -537,6 +549,77 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
         }
     },
 
+    _prepareNextScreen: function(submitPromise, options) {
+        // Make the form completely readonly
+        const form = document.querySelector('form');
+        form.querySelectorAll('input, textarea, label, td')?.forEach(node => {
+            node.classList.add("pe-none");
+        });
+        // Replace the Submit button by a Next button
+        form.querySelector('button[type="submit"]')?.classList.add('d-none');
+        const nextPageBtn = form.querySelector('button[id="next_page"]');
+        nextPageBtn?.classList.remove('d-none');
+        nextPageBtn?.addEventListener('click', () => {
+            this._nextScreen(submitPromise, options);
+        });
+        // On enter or arrow right go to next page
+        const nextPageKeydownListener = (event) => {
+            if (event.code === 'Enter' || event.code === 'ArrowRight') {
+                this._nextScreen(submitPromise, options);
+                // Restore original keydown listener
+                document.removeEventListener('keydown', nextPageKeydownListener);
+                document.addEventListener('keydown', this.documentKeydownListener);
+            }
+        }
+        document.removeEventListener('keydown', this.documentKeydownListener);
+        document.addEventListener('keydown', nextPageKeydownListener);
+    },
+
+    _showCorrectAnswers: function(correctAnswers) {
+        const inactiveQuestionIds = this.options.sessionInProgress ? [] : this._getInactiveConditionalQuestionIds();
+        const activeQuestionIds = Object.keys(correctAnswers).filter(
+            questionId => !inactiveQuestionIds.includes(parseInt(questionId))
+        );
+        activeQuestionIds.forEach(questionId => {
+            const correctAnswer = correctAnswers[questionId];
+            const questionWrapper = document.querySelector(`.js_question-wrapper[id="${questionId}"]`);
+            const answerWrapper = questionWrapper.querySelector('.o_survey_answer_wrapper');
+            const questionType = questionWrapper.querySelector('[data-question-type]').dataset.questionType;
+
+            if (['numerical_box', 'date', 'datetime'].includes(questionType)) {
+                const input = answerWrapper.querySelector('input');
+                let value;
```

Comment:
As you don't reuse value later on, you can directly use:
```suggestion
                let isCorrect;
```

Reply:
I meant directly assigning `isCorrect` in each `if` case too.